libqtile/utils.py

```diff
@@ -209,3 +209,41 @@ def send_notification(title, message, urgent=False, timeout=10000):
         notifier.show()
     except Exception as exception:
         logger.error(exception)
+
+
+def get_default_terminal():
+    """Try to guess terminal."""
+    test_terminals = [
+            ['x-terminal-emulator', '-?'],
+            ['xterm', '-help'],
+            ['urxvt', '-help'],
+            ['st', '-v'],
+            ['gnome-terminal', '-?'],
+            ['terminator', '-v'],
+            ['lxterminal', '-v'],
+            ['mate-terminal', '-h'],
+            ['roxterm', '-?'],
+            ['tilda', '--help'],
+            ['guake', '-h'],
+            ['eterm', '-h'],
+            ['sakura', '-v'],
+            ['konsole', '--help'],
+            ['hyper', '-v'],
+            ['alacritty', '-V'],
+            ]
+
+    to_null = ['>', '/dev/null']
+    for terminal in test_terminals:
+        logger.warning('test '+terminal[0])
+        try:
+            exit_code = os.spawnvp(os.P_WAIT, terminal[0], terminal + to_null)
+            if 127 == exit_code:
+                continue
+        except Exception:
```

Comment:
What kind of exception are we expecting here?

Reply:
I was not sure for 100% here, so I left catcher here. Should I remove?